checkpoint/provider.go

```diff
@@ -91,6 +93,13 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("CHECKPOINT_SESSION_DESCRIPTION", ""),
 				Description: "A description of the session's purpose.",
 			},
+			"session_timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				DefaultFunc:  schema.EnvDefaultFunc("CHECKPOINT_SESSION_TIMEOUT", 600),
+				Description:  "A timeout for the Check Point session in seconds. Can be 10-3600",
+				ValidateFunc: validation.IntBetween(10, 3600),
```

Comment:
We usually don't do validation in the provider because it is already done in the server so you can remove it from here